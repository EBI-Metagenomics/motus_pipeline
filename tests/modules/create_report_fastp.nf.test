nextflow_process {

    name "Test module Fastp report"
    script "modules/create_report_fastp.nf"
    process "FASTP_REPORT"

    test("Report QC (SE)") {

        when {
            params {
                jsons = "${baseDir}/tests/modules/fixtures/fastp_reports/"
                mode = "single"
            }
            process {
                """
                input[0] = channel.fromFilePairs("${params.jsons}/mgtest_fastp.qc.json", checkIfExists: true).map { it[1] }
                input[1] = params.mode
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
    test("Report QC and overlap (PE)") {

        when {
            params {
                jsons = "${baseDir}/tests/modules/fixtures/fastp_reports/"
                mode = "paired"
            }
            process {
                """
                input[0] = channel.fromPath("${params.jsons}/*.json").collect()
                input[1] = params.mode
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
}
