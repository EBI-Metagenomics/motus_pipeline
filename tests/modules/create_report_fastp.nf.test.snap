{
    "Report QC and overlap (PE)": {
        "content": [
            {
                "qc_report": [
                    "qc_summary:md5,74cc08bcc04532c5886e85561bbfe7ef"
                ]
            }
        ],
        "timestamp": "2023-02-28T11:02:35+0000"
    },
    "Report QC (SE)": {
        "content": [
            {
                "qc_report": [
                    "qc_summary:md5,dcbd7acdfe6144da0fcc96ec547414c1"
                ]
            }
        ],
        "timestamp": "2023-02-28T11:02:35+0000"
    }
}